<launch>
  <arg name="ws" default="$(find rsim_kmt)" />
  <arg name="rviz_conf" default="$(find rsim_kmt)/main.rviz" />
  <rosparam command="load" file="$(find rsim_kmt)/config.yaml" />
  <node ns="vcam" pkg="rostopic" name="camera_info" type="rostopic"
      args="pub camera_info sensor_msgs/CameraInfo
     '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'camera'},
      height: 480, width: 640, distortion_model: 'plumb_bob',
      D: [-2.1593156257172657e-02, -4.7221248220475992e-01, 7.4352931622149196e-04, -9.6727275849864641e-04, 2.1955976063477887e+00],
      K: [1.9161277159818778e+03, 0.0, 6.5256256892073213e+02, 0.0,1.9161277159818778e+03, 5.2432464127568198e+02, 0.0, 0.0, 1.0], 
      R: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0],
      P: [1.9161277159818778e+03, 0.0, 6.5256256892073213e+02, 0.0, 0.0, 1.9161277159818778e+03, 5.2432464127568198e+02, 0.0, 0.0, 0.0, 1.0, 0.0],
      binning_x: 0, binning_y: 0,
      roi: {x_offset: 0, y_offset: 0, height: 480, width: 640, do_rectify: false}}' -r 2"
      output="screen" />
  <node pkg="rviz" type="rviz" name="" args="-d $(arg rviz_conf)" />
  <node pkg="rovi" type="floats2pc" name="" args="camera/capture0">
    <remap from="floats" to="/scene/surface/floats" />
    <remap from="pc" to="/scene/surface/pc" />
  </node>
  <node pkg="rovi_utils" type="image_hcat.py" name="" output="screen">
    <remap from="~image_main" to="/rovi/left/image_rect" />
    <remap from="~image_sub" to="/rovi/right/image_rect" />
    <remap from="~image_out" to="/rovi/image_lr" />
    <remap from="~Q" to="/rovi/genpc/Q" />
    <remap from="~K" to="/rovi/left/remap/Kn" />
  </node>
  <node pkg="rovi_utils" type="image_hack.py" name="">
    <remap from="~image_in1" to="/rovi/image_lr" />
    <remap from="~image_in2" to="/gridboard/image_out" />
    <remap from="~image_out" to="/rovi/left/image_rect_mix" />
  </node>
  <node pkg="rovi" type="floats2pc" name="" args="camera/master0">
    <remap from="floats" to="/master/surface/floats" />
    <remap from="pc" to="/master/surface/pc" />
  </node>
  <node pkg="rovi" type="floats2pc" name="" args="camera/capture0/solve0">
    <remap from="floats" to="/master/surface/floats" />
    <remap from="pc" to="/master/surface/pc0" />
  </node>
  <node pkg="rovi" type="floats2pc" name="" args="camera/capture0/solve0_sub">
    <remap from="floats" to="/master/surface_sub/floats" />
    <remap from="pc" to="/master/surface_sub/pc0" />
  </node>
  <node pkg="rtk_tools" type="report.py" name="" output="screen">
    <remap from="/report/update" to="/request/capture" />
    <remap from="/report/complete" to="/response/solve" />
  </node>
</launch>
